test/let_value_test.cpp

```diff
@@ -264,6 +264,20 @@ TEST(Let, SimpleLetValueErrorWithAllocate) {
   })), std::invalid_argument);
 }
 
+TEST(Let, LetValueSuccessorWithException) {
+  EXPECT_THROW(sync_wait(unifex::just() | unifex::let_value([]() {
+    throw std::runtime_error("Throwing error for testing purposes");
+    return unifex::just();
```

Comment:
I'm a little surprised that the compiler doesn't complain this is dead code….

Reply:
Yeah, it won't compile without it :/